### PR TITLE
PV validation range

### PR DESCRIPTION
**Pull request by amecca**
#### PR description:
Small changes in PV validation plots with the TkAl New All-In-One tool to allow fine grained control over the axis range for d<sub>xy</sub> and d<sub>z</sub> vs. p<sub>T</sub>.

#### PR validation:
I ran the tool locally and the plots were presented at the Tracker DPG meeting.

### CODE REVIEW

File: Alignment/OfflineValidation/python/TkAlAllInOneTool/primaryVertexValidationTemplates.py [hunk 1]
```diff
@@ -298,18 +298,26 @@ def switchClusterizerParameters(da):
...
                       .oO[w_dzEtaMax]Oo.,          // width of dz  vs Eta
+                      .oO[w_dxyPtMax]Oo.,          // width of dxy vs Pt
+                      .oO[w_dzPtMax]Oo.,           // width of dz  vs Pt
                       .oO[w_dxyPhiNormMax]Oo.,     // width of dxy vs Phi (norm)
                       .oO[w_dzPhiNormMax]Oo.,      // width of dz  vs Phi (norm)
                       .oO[w_dxyEtaNormMax]Oo.,     // width of dxy vs Eta (norm)
-                      .oO[w_dzEtaNormMax]Oo.       // width of dz  vs Eta (norm)
+                      .oO[w_dzEtaNormMax]Oo.,      // width of dz  vs Eta (norm)
+                      .oO[w_dxyPtNormMax]Oo.,      // width of dxy vs Pt  (norm)
+                      .oO[w_dzPtNormMax]Oo.        // width of dz  vs Pt  (norm)
```
Comment (mmusich) [inline Alignment/OfflineValidation/python/TkAlAllInOneTool/primaryVertexValidationTemplates.py:320]: this file AFAIK is discontinued (even though updating it is harmless).

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Alignment/OfflineValidation/macros/FitPVResiduals.C [hunk 2]
```diff
@@ -4277,18 +4333,22 @@ params::measurement getTheRangeUser(TH1F *thePlot, Limits *lims, bool tag)
   } else {
     if (theTitle.Contains("means")) {
       if (theTitle.Contains("dxy") || theTitle.Contains("dx") || theTitle.Contains("dy")) {
-        if (theTitle.Contains("phi") || theTitle.Contains("pT") || theTitle.Contains("ladder")) {
+        if (theTitle.Contains("phi") || theTitle.Contains("ladder")) {
           result = std::make_pair(-lims->get_dxyPhiMax().first, lims->get_dxyPhiMax().first);
         } else if (theTitle.Contains("eta") || theTitle.Contains("mod")) {
           result = std::make_pair(-lims->get_dxyEtaMax().first, lims->get_dxyEtaMax().first);
+        } else if (theTitle.Contains("pt")) {
```
Comment (mmusich) [inline Alignment/OfflineValidation/macros/FitPVResiduals.C:4340]: shouldn't this be (here and elsewhere)?

```suggestion
        } else if (theTitle.Contains("pT")) {
```

I think `TString::Contains` is case-sensitive.

Comment (amecca) [reply Alignment/OfflineValidation/macros/FitPVResiduals.C:4340]: There is a `ToLower()` ~80 lines above, at the beginning of the function `getTheRangeUser`.
https://github.com/cms-sw/cmssw/blob/a39a770a2cdf573772b1001b295c4db0a2cae520/Alignment/OfflineValidation/macros/FitPVResiduals.C#L4261
I think that it modifies `theTitle`, and the comparison should be correct.
I forgot to mention it in the PR description.

Comment (mmusich) [reply Alignment/OfflineValidation/macros/FitPVResiduals.C:4340]: ah yes, I added that 7 years ago: https://github.com/cms-sw/cmssw/commit/9c4c3330ee1ee0995272710c07300b08ee69efd3

Nonetheless can you add the newly added boundaries to the comment just above?

https://github.com/cms-sw/cmssw/blob/a39a770a2cdf573772b1001b295c4db0a2cae520/Alignment/OfflineValidation/macros/FitPVResiduals.C#L4263-L4282

Comment (amecca) [reply Alignment/OfflineValidation/macros/FitPVResiduals.C:4348]: Done

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+alca
- The relval input error is already present in CMSSW_13_3 2023-10-11-2300, clearly unrelated to this PR, and already fixed in the following IBs

**Comment by rappoccio** [comment]
+1

**Comment by rappoccio** [comment]
merge

* Errors are unrelated.

**Pull request merged**